src/goLanguageServer.ts

```diff
@@ -379,3 +393,96 @@ function registerUsualProviders(ctx: vscode.ExtensionContext) {
 	ctx.subscriptions.push(vscode.languages.registerRenameProvider(GO_MODE, new GoRenameProvider()));
 	vscode.workspace.onDidChangeTextDocument(parseLiveFile, null, ctx.subscriptions);
 }
+
+async function shouldUpdateLanguageServer(tool: Tool, path: string): Promise<boolean> {
+	// Only support updating gopls for now.
+	if (tool.name !== 'gopls') {
+		return false;
+	}
+	// First, run the "gopls version" command and parse its results.
+	// If "gopls" is so old that it doesn't have the "gopls version" command,
+	// or its version doesn't match our expectations, prompt the user to download.
+	const usersVersion = await goplsVersion(path);
+	if (!usersVersion) {
+		return true;
+	}
+	// We might have a developer version. Don't make the user update.
+	if (usersVersion === '(devel)') {
+		return false;
+	}
+	// If the user has a version of gopls that we understand,
+	// ask the proxy for the latest version, and if the user's version is older,
+	// prompt them to update.
+	const listURL = `https://proxy.golang.org/${tool.importPath}/@v/list`;
+	const data = await WebRequest.json<string>(listURL);
+
+	// Coerce the versions into SemVers so that they can be sorted correctly.
+	const versions = [];
+	for (const version of data.trim().split(os.EOL)) {
+		versions.push(semver.coerce(version));
+	}
+	if (versions.length === 0) {
+		return false;
+	}
+	versions.sort();
```

Comment:
This wouldn't work that well... We will have to pass a compareFn to sort()
While 1.12 would come after 1.11, 1.9 will also come after 1.11

Reply:
Something like the below
```javascript
versions.sort((a,b) => {
    if (semver.lt(a,b)) {
        return -1;
    }
    if (semver.gt(a,b)) {
        return 1;
    }
    return 0;
});
```

Reply:
Done. Sorry, I must have misread some documentation and assumed that the correct comparison would be applied automatically in the sort.